### PR TITLE
[REVIEW] Removing nulls on key columns before joining

### LINKED ISSUE #499: [BUG] Join gives wrong output for dataframes with nulls
**Describe the bug**
Blazingsql join behaves differently than SQL on dataframes with nulls.

**Steps/Code to reproduce bug**
Given the following files:

nation1.csv
```
1|Peru|1
2|Bolivia|1
3|Chile|1
4|Ecuador|1
5|USA|2
6|Canada|2
|Arequipa|
```

region1.csv
```
0|Antartica
1|SouthAmerica
2|NorthAmerica
|Euroasia
|PanAmerica
```
Run the query below:
```
from blazingsql import BlazingContext
bc = BlazingContext()

nationColNames= ['n_nationkey', 'n_name', 'n_regionkey']
nationColTypes= ['int32', 'str', 'int32']

regionColNames= ['r_regionkey', 'r_name']
regionColTypes= ['int32', 'str']

region = bc.create_table('region', '/home/rommel/blazingsql_workspace/scripts/region1.csv', fileFormat="csv", delimiter='|', dtype=regionColTypes, names=regionColNames)
nation = bc.create_table('nation', '/home/rommel/blazingsql_workspace/scripts/nation1.csv', fileFormat="csv", delimiter='|', dtype=nationColTypes, names=nationColNames)

sql = "select * from nation n inner join region r on n.n_regionkey=r.r_regionkey"
result_gdf = bc.sql(sql)
print(result_gdf)
```
Current output:
```
BlazingContext ready
  n_nationkey    n_name n_regionkey r_regionkey        r_name
0           1      Peru           1           1  SouthAmerica
1           2   Bolivia           1           1  SouthAmerica
2           3     Chile           1           1  SouthAmerica
3           4   Ecuador           1           1  SouthAmerica
4           5       USA           2           2  NorthAmerica
5           6    Canada           2           2  NorthAmerica
6        null  Arequipa        null        null      Euroasia
7        null  Arequipa        null        null    PanAmerica
```
**Expected behavior**
Expected output:
```
BlazingContext ready
   n_nationkey   n_name  n_regionkey  r_regionkey        r_name
0            1     Peru            1            1  SouthAmerica
1            2  Bolivia            1            1  SouthAmerica
2            3    Chile            1            1  SouthAmerica
3            4  Ecuador            1            1  SouthAmerica
4            5      USA            2            2  NorthAmerica
5            6   Canada            2            2  NorthAmerica
```
**Environment overview**
 - Environment location: Bare-metal
 - Method of cuDF install: conda nightly branch-0.13
 - Method of blazingsql install: sources branch-0.13

### PR DESCRIPTION
**Pull request by rommelDB**
Fixes #499

### CODE REVIEW

**Comment by kharoc** [comment]
CON NULOS

1 RAL

*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1153/1262
FAILED: 109/1262
CRASH: 0/1262
TOTAL: 1262
MAX DELTA: 170.0
*******************************************************************************
Aggregations without group by:   Start Mem: 411.1875   End Mem: 475.1875   Diff: 64.0
Coalesce:   Start Mem: 475.1875   End Mem: 529.1875   Diff: 54.0
Column Basis:   Start Mem: 529.1875   End Mem: 521.1875   Diff: -8.0
Common Table Expressions:   Start Mem: 521.1875   End Mem: 465.1875   Diff: -56.0
Count without group by:   Start Mem: 465.1875   End Mem: 493.1875   Diff: 28.0
Date:   Start Mem: 493.1875   End Mem: 415.1875   Diff: -78.0
Timestamp:   Start Mem: 415.1875   End Mem: 487.1875   Diff: 72.0
Full outer join:   Start Mem: 487.1875   End Mem: 487.1875   Diff: 0.0
Group by:   Start Mem: 487.1875   End Mem: 477.1875   Diff: -10.0
Group by without aggregations:   Start Mem: 477.1875   End Mem: 447.1875   Diff: -30.0
Inner join:   Start Mem: 447.1875   End Mem: 509.1875   Diff: 62.0
Left outer join:   Start Mem: 509.1875   End Mem: 509.1875   Diff: 0.0
Non-EquiJoin Queries:   Start Mem: 509.1875   End Mem: 505.1875   Diff: -4.0
Nested Queries:   Start Mem: 505.1875   End Mem: 427.1875   Diff: -78.0
Order by:   Start Mem: 427.1875   End Mem: 525.1875   Diff: 98.0
Predicates With Nulls:   Start Mem: 525.1875   End Mem: 451.1875   Diff: -74.0
Simple String:   Start Mem: 451.1875   End Mem: 515.1875   Diff: 64.0
Unary ops:   Start Mem: 607.1875   End Mem: 605.1875   Diff: -2.0
Unify Tables:   Start Mem: 605.1875   End Mem: 605.1875   Diff: 0.0
Union:   Start Mem: 605.1875   End Mem: 619.1875   Diff: 14.0
Limit:   Start Mem: 619.1875   End Mem: 453.1875   Diff: -166.0
Where clause:   Start Mem: 453.1875   End Mem: 523.1875   Diff: 70.0
Bindable Alias:   Start Mem: 523.1875   End Mem: 425.1875   Diff: -98.0
Case:   Start Mem: 425.1875   End Mem: 491.1875   Diff: 66.0
Cast:   Start Mem: 491.1875   End Mem: 513.1875   Diff: 22.0
Concat:   Start Mem: 513.1875   End Mem: 529.1875   Diff: 16.0
Literal:   Start Mem: 529.1875   End Mem: 517.1875   Diff: -12.0
Dir:   Start Mem: 517.1875   End Mem: 665.1875   Diff: 148.0
File System Local:   Start Mem: 665.1875   End Mem: 551.1875   Diff: -114.0
File System S3:   Start Mem: 551.1875   End Mem: 589.1875   Diff: 38.0
File System Google Storage :   Start Mem: 589.1875   End Mem: 489.1875   Diff: -100.0
Like:   Start Mem: 489.1875   End Mem: 659.1875   Diff: 170.0
Substring:   Start Mem: 595.1875   End Mem: 577.1875   Diff: -18.0
Wild Card:   Start Mem: 577.1875   End Mem: 555.1875   Diff: -22.0
TPCH Queries:   Start Mem: 555.1875   End Mem: 473.1875   Diff: -82.0
Round:   Start Mem: 473.1875   End Mem: 559.1875   Diff: 86.0


SIN NULOS

1 RAL

========================================================
FAILED TESTS
========================================================
                         TestGroup InputType Result                                           TestId
0   Bindable Alias                  orc       Fail   TEST_09                                        
1   Date                            orc       Fail   TEST_01                                        
2   Inner join                      gdf       Fail   TEST_02                                        
3   Inner join                      orc       Fail   TEST_02                                        
4   Inner join                      parquet   Fail   TEST_02                                        
5   Inner join                      psv       Fail   TEST_02                                        
6   Nested Queries                  gdf       Fail   TEST_01,TEST_03                                
7   Nested Queries                  orc       Fail   TEST_01,TEST_03                                
8   Nested Queries                  parquet   Fail   TEST_01,TEST_03                                
9   Nested Queries                  psv       Fail   TEST_01,TEST_03                                
10  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
11  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
12  Tables from Pandas              gdf       Fail   TEST_14                                        
13  Timestamp                       orc       Fail   TEST_02                                        
14  Unify Tables                    gdf       Fail   TEST_02                                        
15  Unify Tables                    orc       Fail   TEST_02                                        
16  Unify Tables                    parquet   Fail   TEST_02                                        
17  Unify Tables                    psv       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1246/1276
FAILED: 30/1276
CRASH: 0/1276
TOTAL: 1276
MAX DELTA: 180.0
*******************************************************************************
Aggregations without group by:   Start Mem: 425.875   End Mem: 489.875   Diff: 64.0
Coalesce:   Start Mem: 489.875   End Mem: 461.875   Diff: -28.0
Column Basis:   Start Mem: 461.875   End Mem: 553.875   Diff: 92.0
Common Table Expressions:   Start Mem: 553.875   End Mem: 469.875   Diff: -84.0
Count without group by:   Start Mem: 469.875   End Mem: 479.875   Diff: 10.0
Date:   Start Mem: 479.875   End Mem: 451.875   Diff: -28.0
Timestamp:   Start Mem: 451.875   End Mem: 531.875   Diff: 80.0
Full outer join:   Start Mem: 531.875   End Mem: 531.875   Diff: 0.0
Group by:   Start Mem: 531.875   End Mem: 519.875   Diff: -12.0
Group by without aggregations:   Start Mem: 519.875   End Mem: 479.875   Diff: -40.0
Inner join:   Start Mem: 479.875   End Mem: 483.875   Diff: 4.0
Left outer join:   Start Mem: 483.875   End Mem: 483.875   Diff: 0.0
Non-EquiJoin Queries:   Start Mem: 483.875   End Mem: 477.875   Diff: -6.0
Nested Queries:   Start Mem: 477.875   End Mem: 435.875   Diff: -42.0
Order by:   Start Mem: 435.875   End Mem: 477.875   Diff: 42.0
Predicates With Nulls:   Start Mem: 477.875   End Mem: 469.875   Diff: -8.0
Simple String:   Start Mem: 469.875   End Mem: 553.875   Diff: 84.0
Unary ops:   Start Mem: 621.875   End Mem: 615.875   Diff: -6.0
Unify Tables:   Start Mem: 615.875   End Mem: 613.875   Diff: -2.0
Union:   Start Mem: 613.875   End Mem: 597.875   Diff: -16.0
Limit:   Start Mem: 597.875   End Mem: 465.875   Diff: -132.0
Where clause:   Start Mem: 465.875   End Mem: 505.875   Diff: 40.0
Bindable Alias:   Start Mem: 505.875   End Mem: 481.875   Diff: -24.0
Boolean:   Start Mem: 481.875   End Mem: 479.875   Diff: -2.0
Case:   Start Mem: 479.875   End Mem: 487.875   Diff: 8.0
Cast:   Start Mem: 487.875   End Mem: 443.875   Diff: -44.0
Concat:   Start Mem: 443.875   End Mem: 551.875   Diff: 108.0
Literal:   Start Mem: 551.875   End Mem: 525.875   Diff: -26.0
Dir:   Start Mem: 525.875   End Mem: 655.875   Diff: 130.0
File System Local:   Start Mem: 655.875   End Mem: 543.875   Diff: -112.0
File System S3:   Start Mem: 543.875   End Mem: 609.875   Diff: 66.0
File System Google Storage :   Start Mem: 609.875   End Mem: 549.875   Diff: -60.0
Like:   Start Mem: 549.875   End Mem: 729.875   Diff: 180.0
Substring:   Start Mem: 525.875   End Mem: 593.875   Diff: 68.0
Wild Card:   Start Mem: 593.875   End Mem: 563.875   Diff: -30.0
TPCH Queries:   Start Mem: 563.875   End Mem: 547.875   Diff: -16.0
Round:   Start Mem: 547.875   End Mem: 521.875   Diff: -26.0

2 Worker's

*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1497/1510
FAILED: 13/1510
CRASH: 0/1510
TOTAL: 1510
MAX DELTA: 489.0
*******************************************************************************
Aggregations without group by:   Start Mem: 840.875   End Mem: 1329.875   Diff: 489.0
Coalesce:   Start Mem: 1329.875   End Mem: 1273.875   Diff: -56.0
Common Table Expressions:   Start Mem: 1273.875   End Mem: 1275.875   Diff: 2.0
Count without group by:   Start Mem: 1275.875   End Mem: 1273.875   Diff: -2.0
Date:   Start Mem: 1273.875   End Mem: 1315.875   Diff: 42.0
Timestamp:   Start Mem: 1315.875   End Mem: 1319.875   Diff: 4.0
Full outer join:   Start Mem: 1319.875   End Mem: 1317.875   Diff: -2.0
Group by:   Start Mem: 1317.875   End Mem: 1345.875   Diff: 28.0
Group by without aggregations:   Start Mem: 1345.875   End Mem: 1317.875   Diff: -28.0
Inner join:   Start Mem: 1317.875   End Mem: 1359.875   Diff: 42.0
Left outer join:   Start Mem: 1359.875   End Mem: 1325.875   Diff: -34.0
Non-EquiJoin Queries:   Start Mem: 1325.875   End Mem: 1465.875   Diff: 140.0
Nested Queries:   Start Mem: 1465.875   End Mem: 1359.875   Diff: -106.0
Order by:   Start Mem: 1359.875   End Mem: 1437.875   Diff: 78.0
Predicates With Nulls:   Start Mem: 1437.875   End Mem: 1361.875   Diff: -76.0
Simple String:   Start Mem: 1361.875   End Mem: 1375.875   Diff: 14.0
Unary ops:   Start Mem: 1549.875   End Mem: 1543.875   Diff: -6.0
Unify Tables:   Start Mem: 1543.875   End Mem: 1469.875   Diff: -74.0
Union:   Start Mem: 1469.875   End Mem: 1461.875   Diff: -8.0
Limit:   Start Mem: 1461.875   End Mem: 1465.875   Diff: 4.0
Where clause:   Start Mem: 1465.875   End Mem: 1301.875   Diff: -164.0
Bindable Alias:   Start Mem: 1301.875   End Mem: 1385.875   Diff: 84.0
Boolean:   Start Mem: 1385.875   End Mem: 1343.875   Diff: -42.0
Case:   Start Mem: 1343.875   End Mem: 1293.875   Diff: -50.0
Cast:   Start Mem: 1293.875   End Mem: 1309.875   Diff: 16.0
Concat:   Start Mem: 1309.875   End Mem: 1397.875   Diff: 88.0
Literal:   Start Mem: 1397.875   End Mem: 1371.875   Diff: -26.0
Dir:   Start Mem: 1371.875   End Mem: 1547.875   Diff: 176.0
File System Local:   Start Mem: 1547.875   End Mem: 1507.875   Diff: -40.0
File System S3:   Start Mem: 1507.875   End Mem: 1391.875   Diff: -116.0
File System Google Storage :   Start Mem: 1391.875   End Mem: 1503.875   Diff: 112.0
Like:   Start Mem: 1503.875   End Mem: 1587.875   Diff: 84.0
Substring:   Start Mem: 1489.875   End Mem: 1401.875   Diff: -88.0
Wild Card:   Start Mem: 1401.875   End Mem: 1427.875   Diff: 26.0
TPCH Queries:   Start Mem: 1427.875   End Mem: 1461.875   Diff: 34.0
Round:   Start Mem: 1461.875   End Mem: 1381.875   Diff: -80.0

File: engine/src/execution_graph/logic_controllers/LogicalFilter.cpp [hunk 1]
```diff
@@ -410,16 +410,23 @@ std::unique_ptr<ral::frame::BlazingTable> processJoin(
   }
 
   if(join_type == INNER_JOIN) {
+    //Removing nulls on key columns before joining
+    auto table_left_dropna = cudf::experimental::drop_nulls(table_left.view(), left_column_indices);
```
Comment (williamBlazing) [inline engine/src/execution_graph/logic_controllers/LogicalFilter.cpp:414]: We need to check if any of the key columns have any nulls before calling `cudf::experimental::drop_nulls` and only call that funcion if necessary

Comment (rommelDB) [reply engine/src/execution_graph/logic_controllers/LogicalFilter.cpp]: I updated the code, please, check it again.

**Comment by kharoc** [comment]
CON NULOS

1 RAL

*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1194/1278
FAILED: 84/1278
CRASH: 0/1278
TOTAL: 1278
MAX DELTA: 220.0
*******************************************************************************
Aggregations without group by:   Start Mem: 425.875   End Mem: 487.875   Diff: 62.0
Coalesce:   Start Mem: 487.875   End Mem: 437.875   Diff: -50.0
Column Basis:   Start Mem: 437.875   End Mem: 431.875   Diff: -6.0
Common Table Expressions:   Start Mem: 431.875   End Mem: 501.875   Diff: 70.0
Count without group by:   Start Mem: 501.875   End Mem: 531.875   Diff: 30.0
Date:   Start Mem: 531.875   End Mem: 471.875   Diff: -60.0
Timestamp:   Start Mem: 471.875   End Mem: 541.875   Diff: 70.0
Full outer join:   Start Mem: 541.875   End Mem: 541.875   Diff: 0.0
Group by:   Start Mem: 541.875   End Mem: 463.875   Diff: -78.0
Group by without aggregations:   Start Mem: 463.875   End Mem: 555.875   Diff: 92.0
Inner join:   Start Mem: 555.875   End Mem: 459.875   Diff: -96.0
Left outer join:   Start Mem: 459.875   End Mem: 459.875   Diff: 0.0
Non-EquiJoin Queries:   Start Mem: 459.875   End Mem: 443.875   Diff: -16.0
Nested Queries:   Start Mem: 443.875   End Mem: 499.875   Diff: 56.0
Order by:   Start Mem: 499.875   End Mem: 441.875   Diff: -58.0
Predicates With Nulls:   Start Mem: 441.875   End Mem: 473.875   Diff: 32.0
Simple String:   Start Mem: 473.875   End Mem: 499.875   Diff: 26.0
Unary ops:   Start Mem: 581.875   End Mem: 577.875   Diff: -4.0
Unify Tables:   Start Mem: 577.875   End Mem: 577.875   Diff: 0.0
Union:   Start Mem: 577.875   End Mem: 591.875   Diff: 14.0
Limit:   Start Mem: 591.875   End Mem: 433.875   Diff: -158.0
Where clause:   Start Mem: 433.875   End Mem: 535.875   Diff: 102.0
Bindable Alias:   Start Mem: 535.875   End Mem: 477.875   Diff: -58.0
Case:   Start Mem: 477.875   End Mem: 505.875   Diff: 28.0
Cast:   Start Mem: 505.875   End Mem: 527.875   Diff: 22.0
Concat:   Start Mem: 527.875   End Mem: 565.875   Diff: 38.0
Literal:   Start Mem: 565.875   End Mem: 483.875   Diff: -82.0
Dir:   Start Mem: 483.875   End Mem: 703.875   Diff: 220.0
File System Local:   Start Mem: 703.875   End Mem: 627.875   Diff: -76.0
File System S3:   Start Mem: 627.875   End Mem: 629.875   Diff: 2.0
File System Google Storage :   Start Mem: 629.875   End Mem: 627.875   Diff: -2.0
Like:   Start Mem: 627.875   End Mem: 641.875   Diff: 14.0
Substring:   Start Mem: 615.875   End Mem: 505.875   Diff: -110.0
Wild Card:   Start Mem: 505.875   End Mem: 577.875   Diff: 72.0
TPCH Queries:   Start Mem: 577.875   End Mem: 557.875   Diff: -20.0
Round:   Start Mem: 557.875   End Mem: 531.875   Diff: -26.0

**Pull request merged**